runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +162,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TransactionAbortedException) {
```

Comment:
@no2chem I put it in the CorfuCompileProxy because I didn't want to duplicate code in each TransactionalContext implementation (i.e. both snapshot and optimistic). Are you suggesting code duplication code here? or are you thinking of putting this in AbstractTransactionalContext?
If it's the latter, it will look like the xxImpl methods in AbstractQueuedStreamView. What do you think?